### PR TITLE
Move geometry to open3d::t

### PR DESCRIPTION
**Pull request by yxlao**
io will be next, to keep this PR small

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/2403)

### CODE REVIEW

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- cpp/pybind/t/geometry/geometry.h  1
- cpp/open3d/t/geometry/Geometry.h  1
- cpp/tests/t/geometry/TensorListMap.cpp  9
- cpp/tests/t/geometry/PointCloud.cpp  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=6262691&bid=20392487)

**Review by ssheorey — APPROVED** [review]
For consistency, can we move:
-  `cpp/open3d/io/T*` -> `cpp/open3d/io/t/*`
-  `python/test/tgeometry/*` -> `python/test/t/geometry/*`
-  `cpp/tests/io/T*` -> `cpp/tests/io/t/*`

**Comment by germanros1987** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- cpp/open3d/t/geometry/Geometry.h  1
- cpp/tests/t/geometry/TensorListMap.cpp  9
- cpp/pybind/t/geometry/geometry.h  1
- cpp/tests/t/geometry/PointCloud.cpp  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/intel-isl/Open3D/pullRequest?prid=6262691&bid=20392487)

File: cpp/open3d/io/TPointCloudIO.h [hunk 1]
```diff
@@ -29,14 +29,14 @@
...
-#include "open3d/tgeometry/PointCloud.h"
+#include "open3d/t/geometry/PointCloud.h"
 
 namespace open3d {
 namespace io {
 
 /// Factory function to create a pointcloud from a file
 /// Return an empty pointcloud if fail to read the file.
-std::shared_ptr<tgeometry::PointCloud> CreatetPointCloudFromFile(
+std::shared_ptr<t::geometry::PointCloud> CreatetPointCloudFromFile(
```
[the remark this answers is not in the document]
Comment (yxlao) [reply cpp/open3d/io/TPointCloudIO.h:39]: yes

**Comment by yxlao** [comment]
> For consistency, can we move:
> -  `cpp/open3d/io/T*` -> `cpp/open3d/io/t/*`
> -  `python/test/tgeometry/*` -> `python/test/t/geometry/*`
> -  `cpp/tests/io/T*` -> `cpp/tests/io/t/*`

@ssheorey This is done in https://github.com/intel-isl/Open3D/pull/2406

**Pull request merged**